mbedtls/src/pk/mod.rs

```diff
@@ -295,16 +246,18 @@ impl Pk {
         Ok(ret)
     }
 
-    pub fn private_from_ec_components(mut curve: EcGroup, private_key: Mpi) -> Result<Pk> {
+    // This function now need a RNG because it utilized `EcPoint::mul` function which must need
+    // a random number generator for blinding in mbedtls 3.X, 
+    pub fn private_from_ec_components<F: Random>(rng: &mut F, mut curve: EcGroup, private_key: Mpi) -> Result<Pk> {
```

Comment:
Same applies here. `rng` is only needed when computing public data from private bytes.